2nd-gen/packages/swc/components/asset/Asset.ts

```diff
@@ -11,62 +11,89 @@
         ></path>
         <path
-            class="fileOutline"
+            class="spectrum-Asset-fileOutline"
             d="M113.1,23.3L92.6,2.9C90.7,1,88.2,0,85.5,0H24c-6.6,0-12,5.4-12,12v104c0,6.6,5.4,12,12,12h80c6.6,0,12-5.4,12-12V30.4C116,27.8,114.9,25.2,113.1,23.3z M90,6l20.1,20H92c-1.1,0-2-0.9-2-2V6z M112,116c0,4.4-3.6,8-8,8H24c-4.4,0-8-3.6-8-8V12c0-4.4,3.6-8,8-8h61.5c0.2,0,0.3,0,0.5,0v20c0,3.3,2.7,6,6,6h20c0,0.1,0,0.3,0,0.4V116z"
         ></path>
     </svg>
 `;
 
 const folder = (label: string): TemplateResult => html`
     <svg
-        class="folder"
+        class="spectrum-Asset-folder"
         role="img"
         viewBox="0 0 32 32"
         aria-label=${label || 'Folder'}
     >
         <path
-            class="folderBackground"
+            class="spectrum-Asset-folderBackground"
             d="M3,29.5c-1.4,0-2.5-1.1-2.5-2.5V5c0-1.4,1.1-2.5,2.5-2.5h10.1c0.5,0,1,0.2,1.4,0.6l3.1,3.1c0.2,0.2,0.4,0.3,0.7,0.3H29c1.4,0,2.5,1.1,2.5,2.5v18c0,1.4-1.1,2.5-2.5,2.5H3z"
         ></path>
         <path
-            class="folderOutline"
+            class="spectrum-Asset-folderOutline"
             d="M29,6H18.3c-0.1,0-0.2,0-0.4-0.2l-3.1-3.1C14.4,2.3,13.8,2,13.1,2H3C1.3,2,0,3.3,0,5v22c0,1.6,1.3,3,3,3h26c1.7,0,3-1.4,3-3V9C32,7.3,30.7,6,29,6z M31,27c0,1.1-0.9,2-2,2H3c-1.1,0-2-0.9-2-2V7h28c1.1,0,2,0.9,2,2V27z"
         ></path>
     </svg>
 `;
 
 /**
+ * Use an asset element to visually represent a file, folder, or image.
+ * File and folder representations center themselves within the available space.
+ * Images are contained to the element’s size and centered.
+ *
  * @element swc-asset
- * @slot - content to be displayed in the asset when an acceptable value for `file` is not present
+ * @slot - content to be displayed when no `variant` is set (typically an <img>)
+ *
+ * @example
+ * <swc-asset style="block-size: 128px">
+ *   <img class="spectrum-Asset-image" src="example.png" alt="Example image" />
+ * </swc-asset>
+ *
+ * @example
+ * <swc-asset variant="file" style="min-inline-size: 150px; block-size: 128px"></swc-asset>
+ *
+ * @example
+ * <swc-asset variant="folder" style="min-inline-size: 150px; block-size: 128px"></swc-asset>
  */
 export class Asset extends AssetBase {
+    // ──────────────────────────────
+    //     RENDERING & STYLING
+    // ──────────────────────────────
+
     public static override get styles(): CSSResultArray {
         return [styles];
     }
 
     protected override render(): TemplateResult {
-        if (this.variant === 'file') {
-            return file(this.label);
-        } else if (this.variant === 'folder') {
-            return folder(this.label);
-        }
-        return html` <slot></slot> `;
+        return html`
+            <div
+                class=${classMap({
+                    ['spectrum-Asset']: true,
+                })}
```

Comment:
only one static class is set you could just set `class="spectrum-Asset"` directly. minor simplification.

Reply:
i would like to keep the classMap pattern so its easily extensible if it arises... not a deal breaker though